### PR TITLE
feat(start-stack): add flag to start stack from main branch of DSP-API repository (DEV-2396)

### PR DESCRIPTION
**Pull request by jnussbaum**

### CODE REVIEW

File: src/dsp_tools/utils/stack_handling.py [hunk 1]
```diff
@@ -219,17 +252,24 @@ def _start_remaining_docker_containers(self) -> None:
...
         """
-        subprocess.run("docker compose up -d", shell=True, cwd=self.docker_path_of_user, check=True)
+        if self.__stack_configuration.latest_dev_version:
+            subprocess.run(
+                "docker pull daschswiss/knora-api:latest",
+                shell=True,
+                cwd=self.__docker_path_of_user,
+                check=True,
+            )
+        subprocess.run("docker compose up -d", shell=True, cwd=self.__docker_path_of_user, check=True)
```
Comment (jnussbaum) [inline src/dsp_tools/utils/stack_handling.py:262]: This is one of the relevant code changes

File: src/dsp_tools/utils/stack_handling.py [hunk 2]
```diff
@@ -255,39 +295,46 @@ def _start_docker_containers(self) -> None:
...
+
+        with open(docker_compose_file, "r", encoding="utf-8") as f:
+            docker_compose_text = yaml.safe_load(f)
+        old_tag = docker_compose_text.get("services", {}).get("api", {}).get("image", "")
+        if not re.search(r"daschswiss/knora-api:\d+\.\d+\.\d+", old_tag):
+            raise BaseError(f"Invalid tag '{old_tag}' in docker-compose.yml > services > api > image")
+
+        docker_compose_text["services"]["api"]["image"] = "daschswiss/knora-api:latest"
+        with open(docker_compose_file, "w", encoding="utf-8") as f:
+            yaml.dump(docker_compose_text, f)
```
Comment (jnussbaum) [inline src/dsp_tools/utils/stack_handling.py:320]: This is one of the relevant code changes

**Review by BalduinLandolt — COMMENTED** [review]
See my comment

File: src/dsp_tools/utils/stack_handling.py (hunk 2 above)
Comment (BalduinLandolt) [reply src/dsp_tools/utils/stack_handling.py:320]: or you could use an environment variable to control the version: https://stackoverflow.com/a/62706577/11914562

File: src/dsp_tools/dsp_tools.py [hunk 3]
```diff
@@ -243,6 +243,11 @@ def make_parser() -> argparse.ArgumentParser:
     parser_stackup.add_argument(
         "--no-prune", action="store_true", help="don't execute 'docker system prune' (and don't ask)"
     )
+    parser_stackup.add_argument(
+        "--latest-dev-version",
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply src/dsp_tools/dsp_tools.py]: done

**Pull request merged**